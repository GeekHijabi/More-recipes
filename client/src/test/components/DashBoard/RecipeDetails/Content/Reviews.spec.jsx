import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { Reviews }
  from '../../../../../js/components/DashBoard/RecipeDetails/Content/Reviews';

const props = {
  apiDeleteRecipeReview: jest.fn(() => Promise.resolve({ ok: true })),
  recipe: {
    ingredient: '',
    description: '',
  },
  description: 'boil rice',
  activeTab: {
    name: {
      ingredients: 'ingredients',
      description: 'description',
      reviews: 'reviews'
    }
  },
  reviewedRecipe: {
    id: 1,
    recipeName: 'rice',
    description: 'boil rice',
    ingredients: 'rice, water'
  },
  recipeId: 2
};

describe('Component: Reviews', () => {
  beforeEach(() => {
    global.swal = {
      success: () => {},
      error: () => {}
    };
  });
  describe('Review snapshot', () => {
    it('should render correctly', () => {
      const reviews = shallow(<Reviews {...props} />);
      expect(reviews).toMatchSnapshot();
    });
  });
  it('should recieve next props', () => {
    const spy = sinon.spy(Reviews.prototype, 'componentWillReceiveProps');
    const nextProps = {
      apiDeleteRecipeReview: jest.fn(() => Promise.resolve({ ok: true })),
      recipe: {
        ingredient: '',
        description: '',
      },
      description: 'boil rice',
      activeTab: {
        name: {
          ingredients: 'ingredients',
          description: 'description',
          reviews: 'reviews'
        }
      },
      reviewedRecipe: {
        Reviews: [
          {
            id: 2,
            Reviews: 'Nice one',
            User: {
              imageUrl: 'imageUrl',
              userName: 'userName'

            }
          }
        ]
      },
      recipeId: 2
    };

    const wrapper = shallow(<Reviews {...nextProps} />);
    wrapper.setProps({ nextProps });
    expect(spy.calledOnce).toEqual(true);
  });

  describe('onDelete', () => {
    it('should remove the review', () => {
      const wrapper = shallow(<Reviews {...props} />);
      wrapper.instance().onDelete();
      expect(props.apiDeleteRecipeReview.mock.calls.length).toEqual(0);
    });
  });
});
