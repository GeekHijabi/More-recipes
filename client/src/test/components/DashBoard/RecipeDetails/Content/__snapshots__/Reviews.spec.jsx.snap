// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Reviews Review snapshot should render correctly 1`] = `
<div
  aria-labelledby="reviews-tab"
  className="tab-pane fade"
  id="reviews"
  role="tabpanel"
>
  <Connect(AddReviewForm)
    recipeId={2}
  />
  <ul
    className="tab-pane fade"
  >
    <span
      className="reviewText"
    >
      This recipe has not been reviewed yet, Add a review.
    </span>
  </ul>
</div>
`;
